lib/sensors/time_of_flight.hpp

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include "i2c_sensors.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+
+#include <core/logger.hpp>
+#include <io/i2c.hpp>
+
+namespace hyped::sensors {
+constexpr std::uint8_t kDefaultTimeOfFlightAddress = 0x29;
+
+// TODOLater - Implement II2cMuxSensor
+class TimeOfFlight {
+ public:
+  static std::optional<TimeOfFlight> create(core::ILogger &logger,
+                                            std::shared_ptr<io::II2c> i2c,
+                                            const std::uint8_t channel,
+                                            const std::uint8_t device_address
+                                            = kDefaultTimeOfFlightAddress);
+
+  ~TimeOfFlight();
+
+  std::optional<std::uint8_t> getRangeSingleShot();
+
+  std::optional<std::uint8_t> getRangeContinuous();
+
+  // TODOlater - Confirm if this function works/is required
+  std::optional<std::uint8_t> getStatus();
+
+  std::uint8_t getChannel();
+
+ private:
+  TimeOfFlight(core::ILogger &logger,
+               std::shared_ptr<io::II2c> i2c,
+               const std::uint8_t channel,
+               const std::uint8_t device_address);
+
+  /**
+   * @brief Set registers in device for default use
+   * @return kSuccess if successful; kFailure otherwise
+   * @note See ST Application Note AN4545 Section 1.3 for details
+   */
+  core::Result initialise();
+
+  /**
+   * @brief Reads the measured range
+   * @return Range value in millimetres
+   * @note Implementation based on ST Application Note AN4545
+   */
+  std::optional<std::uint8_t> getRange();
+
+  /**
+   * @brief Check if the sensor is in the appropriate mode before trying to read the range, and set
+   * it to mode_value if it isn't
+   * @param mode_value  0b01 or 0b11 for single-shot or continuous
+   * @return kSuccess if the sensor mode is correct, and kFailure otherwise
+   */
+  core::Result checkSensorMode(std::uint8_t mode_value);
+
+ private:
+  // TODOLater - Confirm these addresses are correct
+  // Register addresses/values taken from the VL6180X datasheet
```

Comment:
Not sure where exactly they've been taken from, or what they all mean - Are these needed?